File: src/constants/businessCategories.js
// Business categories for AEO analysis
// Maintains consistency with backend constants in src/constants/business_categories.py

/**
 * Business category configuration for the AEO analysis system.
 * Each category has a value (used in API calls) and a label (displayed in UI).
 * 
 * @typedef {Object} BusinessCategory
 * @property {string} value - The internal value used in API calls and storage
 * @property {string} label - The human-readable label displayed in the UI
 */
export const BUSINESS_CATEGORIES = [
  { value: 'e-commerce', label: 'E-commerce' },
  { value: 'saas', label: 'SaaS' },
  { value: 'local-services', label: 'Local Services' },
  { value: 'healthcare', label: 'Healthcare' },
  { value: 'finance', label: 'Finance' },
  { value: 'education', label: 'Education' },
  { value: 'real-estate', label: 'Real Estate' },
  { value: 'travel-hospitality', label: 'Travel & Hospitality' },
  { value: 'professional-services', label: 'Professional Services' },
  { value: 'manufacturing', label: 'Manufacturing' },
  { value: 'other', label: 'Other' }
];

// Export as array of values for backend validation and consistency checks
export const BUSINESS_CATEGORY_VALUES = BUSINESS_CATEGORIES.map(cat => cat.value);

// Helper function to get label by value
export const getBusinessCategoryLabel = (value) => {
  const category = BUSINESS_CATEGORIES.find(cat => cat.value === value);
  return category ? category.label : value;
};

File: src/constants/business_categories.py
"""
Business categories for AEO analysis.

This module defines the valid business categories that users can select
to tailor their AEO analysis. These categories are used for:
- API validation in Pydantic models
- Frontend dropdown options
- Future prompt generation customization

Maintains consistency with frontend constants in src/constants/businessCategories.js
"""

from typing import List, Literal

# Type definition for better type safety
BusinessCategoryType = Literal[
    'e-commerce',
    'saas', 
    'local-services',
    'healthcare',
    'finance',
    'education',
    'real-estate',
    'travel-hospitality',
    'professional-services',
    'manufacturing',
    'other'
]

# Business categories for AEO analysis
BUSINESS_CATEGORIES: List[str] = [
    'e-commerce',
    'saas',
    'local-services',
    'healthcare',
    'finance',
    'education',
    'real-estate',
    'travel-hospitality',
    'professional-services',
    'manufacturing',
    'other'
]

def is_valid_business_category(category: str) -> bool:
    """
    Check if a given category is valid.
    
    Args:
        category: The category string to validate
        
    Returns:
        True if the category is valid, False otherwise
    """
    return category in BUSINESS_CATEGORIES